tests/phpunit/integration/Core/Notifications/NotificationsTest.php

```diff
@@ -42,4 +47,35 @@ function ( REST_Route $route ) {
 			$routes
 		);
 	}
+
+	public function test_non_proxy_request_does_not_request_proxy() {
+		// Create a user with access to the WP REST API and log in.
+		$user = $this->factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user->ID );
+		do_action( 'wp_login', $user->user_login, $user );
+
+		// Fake setup and authentication for access to dashboard.
+		$this->fake_site_connection();
+		remove_all_filters( 'googlesitekit_setup_complete' );
+		$authentication = new Authentication( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$authentication->verification()->set( true );
+		$authentication->get_oauth_client()->set_access_token( 'test-access-token', HOUR_IN_SECONDS );
+
+		// Set up the REST Routes including the notification route to test.
+		$rest_routes = new REST_Routes( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$rest_routes->register();
+
+		// Get a REST server instance.
+		$server = rest_get_server();
+
+		// Make the test request.
+		$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/core/site/data/notifications' );
+		$response = $server->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals(
+			array(),
+			$data
+		);
```

Comment:
While this does somewhat test the request, it doesn't actually test that the request wasn't made to the proxy which is the most important thing here. Also, it's worth noting that an empty array doesn't specifically mean that a request wasn't made as the proxy could return an empty array as well as a valid response (i.e. no notifications).

In this case, we should simply hook into the `pre_http_request` filter before dispatching the rest request, and if a request is made with a URL containing the proxy's notifications endpoint (`$this->google_proxy->url( '/notifications/' )`), then the test should fail (regardless of the response). We have a few other tests where this filter is used that you could use as a reference, but essentially we want to do something like this:

```php
// in test
add_filter( 'pre_http_request', function ( $override, $args, $url ) {
	if ( /* $url starts with notifications endpoint URL */ ) {
		$this->fail( 'Expected no request to be made to Google proxy notifications endpoint' );
	}
	return $override;
}, 10, 3 );
///
```

The rest of the test can be simplified to meet the minimum requirements needed to make the request and if the response code is a 200 then we know the permissions were met.